### PR TITLE
ENG-3824 configMfe attribute validations for microservices in bundle descriptor

### PR DESCRIPTION
**Pull request by jeffgo10**

### CODE REVIEW

File: src/models/component.ts [hunk 1]
```diff
@@ -24,6 +24,7 @@ export type Component<T extends ComponentType> = {
   name: string
   stack: StackFor<T>
   type: T
+  others?: any
```
Comment (cecchisandrone) [inline src/models/component.ts]: I don't like this, it's too vague. Can't we just map only the props we need explicitly? mfeType, configMfe if I got it well

File: src/services/component-service.ts [hunk 2]
```diff
@@ -133,6 +134,29 @@ export class ComponentService {
...
+        mfe.others?.configMfe &&
+        mfe.others.configMfe.slice(0, 9) !== 'internal:' &&
+        allMfesWithTypes.findIndex(
+          ({ name, type }) =>
+            mfe.others.configMfe === name &&
+            type === MicroFrontendType.WidgetConfig
+        ) === -1
+      ) {
+        throw new CLIError(
+          `Config MFE ${mfe.others.configMfe} must be at least an internal widget or must exist among your micro frontends with type widget-config`
```
Comment (cecchisandrone) [inline src/services/component-service.ts]: `configMfe value ${mfe.others.configMfe} for MFE ${mfe} must be an existing widget-config MFE in the project or a platform provided config MFE, defined with the internal:<name> syntax`

File: src/models/component.ts (hunk 1 above)
Comment (jeffgo10) [reply src/models/component.ts]: done. I removed `others` props and added props `mfeType` and `configMfe` (limited to ComponentType.MICROFRONTEND only)

File: src/services/component-service.ts (hunk 2 above)
Comment (jeffgo10) [reply src/services/component-service.ts]: done

**Review by cecchisandrone — APPROVED** [review]

**Pull request merged**